src/provider/states/create_config.rs

```diff
@@ -0,0 +1,296 @@
+                if let Some(config_map) = &volume.config_map {
+                    // config map was present, check if a name was set
+                    // not sure when it would not be set, but it is a valid possibility, so we need
+                    // to handle it - if no name is present, we'll just ignore this map, not sure
+                    // how to retrieve it otherwise
+                    if let Some(config_map_name) = &config_map.name {
+                        debug!("Found reference to config map {}", &config_map_name);
+                        get_config_maps.push(String::from(config_map_name));
+                    }
+                }
+            }
+        }
+        get_config_maps
+    }
+
+    async fn retrieve_config_map(
+        &self,
+        client: Client,
+        name: String,
+    ) -> Result<ConfigMap, StackableError> {
+        let config_maps: Api<ConfigMap> = Api::namespaced(client.clone(), "default");
+
+        Ok(config_maps.get(&name).await?)
+    }
+
+    fn apply_config_map(
+        &self,
+        map: ConfigMap,
+        target_directory: PathBuf,
+        template_data: &BTreeMap<String, String>,
+    ) -> Result<(), StackableError> {
+        let config_map_name = map.metadata.name.unwrap_or(String::from("undefined"));
+        debug!(
+            "applying configmap {} to directory {:?}",
+            &config_map_name, target_directory
+        );
+        if !(&target_directory.is_dir()) {
+            info!("creating config directory {:?}", target_directory);
+            fs::create_dir_all(&target_directory)?;
+        }
+        if let Some(data) = map.data {
+            debug!("Map contained keys: {:?}", &data.keys());
+            for key in data.keys() {
+                debug!("found key: {} in configmap {}", key, &config_map_name);
+                if let Some(content) = data.get(key) {
+                    trace!("content of key: {}", &content);
+                    debug!("rendering");
+                    let rendered_content = CreatingConfig::render_config_template(
+                        template_data.clone(),
+                        content.clone(),
+                    )?;
+                    debug!("done rendering");
+                    let target_file = target_directory.join(&key);
+
+                    // TODO: compare existing file with intended state
+                    if CreatingConfig::needs_update(&target_file, &rendered_content)? {
+                        debug!(
+                            "writing content of map entry {} to file {:?}",
+                            key, target_file
+                        );
+                        let write_result = fs::write(target_directory.join(&key), rendered_content);
+                        match write_result {
+                            Ok(()) => debug!("write of file {:?} successful!", target_file),
+                            Err(e) => error!("write of file {:?} failed: {}", target_file, e),
```

Comment:
Shouldn't this return an error as well? I think this will "silently" fail ...well with the exception that it's not silent but if it fails nothing will happen, right?

Reply:
It should indeed, added this.